bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/osgi/ResourceBundleClassLoader.java

```diff
@@ -36,10 +36,15 @@
  * mappings.
  *
  * @author Michael Grammling - Initial Contribution
+ * @author Martin Herbst - ISO-8859-1 support added
  */
 public class ResourceBundleClassLoader extends ClassLoader {
 
-    private static final String[] SUPPORTED_CHARSETS = { "UTF-8" };
+    /**
+     * ISO-8859-1 must be the last array element because all files could be decoded to ISO-8859-1 and
+     * therefore always character set ISO-8859-1 would be assumed.
```

Comment:
> Does this mean that decoder.decode() will never throw a CharacterCodingException, since every byte is a valid character in ISO-8859-1? 

Exactly. I am not sure what happens with values below x'20'. But all values that are greater or equal are accepted as valid characters.

> And UTF-8 decoding will fail on any non-ASCII character, because these bytes are not valid in an UTF-8 byte array?

It simple just checks for valid byte combinations. A byte value that is greater then x7F is only allowed if the preceding byte has a special value.

This means that the routine can only detect if a text is *not* UTF-8.

> Would it be possible to add some unit test in org.eclipse.smarthome.core.test that actually tests this charset detection mechanism? I would feel more comfortable with that :-)

I understand :-). I think it is not necessary to test the whole class (would be complicated because of the bundle mechanism) but only the character detection method detectCharset. It is private but with the help of some reflection this will work.